src/lib/recursion_level.rs

```diff
@@ -0,0 +1,24 @@
+//! # recursion_level
+//!
+//! Utility functions to keep track of the recursion of cargo-make calls.
+//!
+
+#[cfg(test)]
+#[path = "./recursion_level_test.rs"]
+mod recursion_level_test;
+
+use envmnt;
+
+static RECURSION_ENV_VAR_NAME: &str = "CARGO_MAKE_INTERNAL_RECURSION";
+
+pub(crate) fn recursion_level() -> u32 {
+    envmnt::get_or(RECURSION_ENV_VAR_NAME, "0").parse().expect("failed to retrieve env var")
+}
+
+pub(crate) fn is_first_level() -> bool {
+    recursion_level() == 0
+}
+
+pub(crate) fn increase_level() {
+    envmnt::set(RECURSION_ENV_VAR_NAME, (recursion_level() + 1).to_string())
```

Comment:
in order to ensure the first cargo-make stays at 0.
i think this function (and tell me if this makes sense), should only increment if value exists. if not, it should set it to 0 and next invocation of it, will increase it to 1.
it would also resolve your issue with the cli.rs that you delayed setting it.